### DISCUSSION TITLE
Interoperability with FastDDS implementation

### DISCUSSION THREAD

**Discussion by AlbertoElDevaluador** [jmmorato/openddsharp#193] [Q&A] [answered]
Should there be problems between DDS communication if I make a subscriber with this library with OpenDDS and the publisher with FastDDS from C++? I should receive everything correctly, right?

**Comment by jmmorato** [comment]
I didn't follow too much, but same questions are active in OpenDDS and FastDDS repos

https://github.com/eProsima/Fast-DDS/issues/3839
https://github.com/OpenDDS/OpenDDS/issues/4268

**Reply by AlbertoElDevaluador** [reply]
i mean that, if a publisher publish a topic using fastDDS, can i receive it with my suscriber made with openddsSharp????

**Reply by jmmorato** [reply, marked as answer]
Seems the issue is just a different version of the serialization used by each implementation.

If you configure your OpenDDSharp DataReaders to use XCDR1 it should work. Something like:

```csharp
var drQos = new DataReaderQos();
drQos.Representation.Value = new List<short> { DataRepresentationQosPolicy.XCDR_DATA_REPRESENTATION };
```

**Reply by AlbertoElDevaluador** [reply]
Thank you!!
